roles/monitor/templates/monitor.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+if test -f {{ ansible_env.HOME }}/ocluster.log ; then
+	kc=$(tail {{ ansible_env.HOME }}/ocluster.log | grep pkill | wc -l)
```

Comment:
`kc` = kill count

> suppose new logging is added that prevents the accumulation of 10 consecutive kill commands, even when it is looping

That would seem highly unlikely unless `kill_users_processes` was edited at which point we'd presumably be editing it to fix this problem and thus not need this script.

> or imagine new logic is added that does a pkill to clean up some other spawned processes, and it accumulates 10 successive kills, but it isn't actually related to a defunct process that requires a restart.

If we've run `pkill` > 10 times then it's not going to work.  We always need the kill to work in order to `umount` the ZFS volume.

> I think a more robust approach might by to check for defunct process belonging to the user ID under we are running the builder process? IIUC, we know that statically, based on

No, we don't want to reboot when we see a defunct process.  The parent may terminate naturally, or via pkill, removing the defunct process and the need to reboot.

Reply:
> That would seem highly unlikely unless kill_users_processes was edited at which point we'd presumably be editing it to fix this problem and thus not need this script.

We actually have two levels of loops that could lead to repeated killp, if I'm reading the code correctly, (see https://github.com/ocurrent/obuilder/blob/6bb4a4e9b1f4134ff0b1ade523427ad06dbab6ed/lib/macos.ml#L39-L60) and the wider context of https://github.com/ocurrent/obuilder/blob/6bb4a4e9b1f4134ff0b1ade523427ad06dbab6ed/lib/sandbox.macos.ml#L84-L102 . While it is probably unlikely, it is not implausible afaict, and I am not sure about the down sides of erring on a less fragile solution. I'm happy to talk it over tho.

> If we've run pkill > 10 times then it's not going to work. We always need the kill to work in order to umount the ZFS volume.

We could be killing 10

> No, we don't want to reboot when we see a defunct process. The parent may terminate naturally, or via pkill, removing the defunct process and the need to reboot.

Sorry, I only meant to refer to orphaned defunct processes (that belonged to the configured user). Since https://github.com/ocurrent/obuilder/blob/6bb4a4e9b1f4134ff0b1ade523427ad06dbab6ed/lib/macos.ml#L39-L60 is issuing `sudo pkill -9`, I am assuming that the only reason it cannot kill the process is that it has defaulted its parent to init. If we think these processes may actually have a live parent, then maybe we should put this on hold and wait until a recurrence so that we can figure out whether or not that is the case. IMO, it would be quite a bit nicer to not need to reboot the machine if we could improve the correctness of obuilder on this instead.